src/robot/libdocpkg/xmlwriter.py

```diff
@@ -93,20 +93,18 @@ def _write_arguments(self, kw, writer):
                                  'repr': unicode(arg)})
             if arg.name:
                 writer.element('name', arg.name)
-            if arg.type is not arg.NOTSET:
-                writer.element('type', arg.type_repr)
+            if arg.types:
+                for type_repr in arg.types_reprs:
+                    writer.element('type', type_repr)
```

Comment:
Isn't `if arg.types` unnecessary? If the arg doesn't have types, `arg.types_reprs` yields nothing and the loop isn't executed.